File: scala/src/main/scala/spark/spark.scala
package spark

import org.apache.spark.{SparkConf, SparkContext}

object spark {
  def main(args: Array[String]) {

    val logFile = "s3a://elasticmapreduce/samples/wordcount/wordSplitter.py"

    val conf = new SparkConf().setAppName("Simple Application")
      .setMaster("local")

    val sc = new SparkContext(conf)

    val logData = sc.textFile(logFile).cache()

    val numAs = logData.filter(line => line.contains("a")).count()
    val numBs = logData.filter(line => line.contains("b")).count()
    println(s"Lines with a: $numAs, Lines with b: $numBs")

    sc.stop()
  }
}

File: src/spark.py
from pyspark import SparkContext, SparkConf


def sparkly():
    logFile = "s3a://elasticmapreduce/samples/wordcount/wordSplitter.py"

    conf = SparkConf()\
        .setAppName("Simple Application")\
        .setMaster("local")

    #sc = SparkContext(conf)                    # for whatever reason, but this will not work
    sc = SparkContext("local", "simpleApp")     # this will however...

    logData = sc.textFile(logFile).cache()

    numAs = logData.filter(lambda line: line.count("a")).count()
    numBs = logData.filter(lambda line: line.count("b")).count()
    print("Lines with a: {0}, Lines with b: {1}".format(numAs, numBs))

    sc.stop()

if __name__ == "__main__":
    sparkly()